stdlib/public/stubs/SwiftNativeNSXXXBase.mm.gyb

```diff
@@ -117,6 +133,37 @@ extern "C" int32_t swift_stdlib_compareNSStringDeterministicUnicodeCollation(
 }
 
 SWIFT_RUNTIME_STDLIB_INTERFACE
+extern "C" int32_t
+swift_stdlib_compareNSStringDeterministicUnicodeCollationStack(
+    void *baseAddressLhs, uintptr_t countAndFlagsLhs, void *ownerLhs,
+    void *baseAddressRhs, uintptr_t countAndFlagsRhs, void *ownerRhs) {
+
+  _SwiftStackAllocatedNSContiguousString stackAllocatedStringLhs = {
+      {
+          &_TMCs19_NSContiguousString, // isa.
+      },
+      baseAddressLhs,
+      countAndFlagsLhs, // core.
+      ownerLhs
+  };
+  _SwiftStackAllocatedNSContiguousString stackAllocatedStringRhs = {
+      {
+          &_TMCs19_NSContiguousString, // isa.
+      },
+      baseAddressRhs,
+      countAndFlagsRhs, // core.
+      ownerRhs
```

Comment:
I see, that's actually surprising for me.  I think exploring the direction where we can do the allocation on the Swift side is going to result in shortest code :)